### ISSUE TITLE
ProcessingSet plot_phase_centers() field labels ignored

### ISSUE THREAD

**Issue by pford** [casangi/xradio#315]
- plot_phase_centers(): **label_all_fields** argument is ignored.

Side note: would it be useful to users to label antenna names in plot_antenna_positions() (or show in hover)?

**Comment by Jan-Willem** [comment]
We will fix it.

**Comment by pford** [comment]
> In this issue, I assume that by label we mean an annotation next to the field center/offset, as opposed to a label in a legend which is what plot_phase_centers() is using. Here are some example screenshots. Let us know what you think and any other suggestions:
> 

@FedeMPouzols I did not see all of the field names in a legend when `label_all_fields=True`.  In the `main` branch, I get the plot below **whether label_all_fields is True or False**.  This is why I said this parameter was ignored.  Sorry, I should have included this plot as an example in the initial issue description.

![Image](https://github.com/user-attachments/assets/44e46d86-edc6-4160-a6cb-bfd1fd8ee2ae)

In your branch, as in your posted plots, the plot has each field labeled on the plot when label_all_fields=True.  Do we need a legend for the center field if it is labelled along with the others?

![Image](https://github.com/user-attachments/assets/a5d817be-dfdc-4f28-bc45-3c1ef2b133bd)

**Comment by FedeMPouzols** [comment]
Yes, I think that when we have the labels/annotations next to the points we don't need the legend with the center (red) field, I'll remove it when `label_all_fields=True`.

**Comment by FedeMPouzols** [comment]
A couple of examples of how the plots look after the last change (do not show legend when label_all_fields, as it is redundant and takes some space from the plot):

![Image](https://github.com/user-attachments/assets/59afa910-e520-46b6-ad23-29c94ccfc32c)

![Image](https://github.com/user-attachments/assets/e6a7d7ba-0eb1-44f2-80b0-1267677c8d31)

When label_all_fields=False, the legend is still shown for the center (red) field:

![Image](https://github.com/user-attachments/assets/43df1ae6-fa87-4ada-824c-a15399487336)

**Comment by pford** [comment]
@FedeMPouzols thank you, that looks great!
